src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilerCache.cs

```diff
@@ -57,17 +57,17 @@ public class CompilerCache : ICompilerCache
                 }
             }
 
-            // Set up ViewStarts
+            // Set up _GlobalImports
             foreach (var entry in cacheEntries)
             {
-                var viewStartLocations = ViewStartUtility.GetViewStartLocations(entry.RelativePath);
-                foreach (var location in viewStartLocations)
+                var globalFileLocations = ViewHierarchyUtility.GetGlobalImportLocations(entry.RelativePath);
+                foreach (var location in globalFileLocations)
                 {
-                    var viewStartEntry = _cache.Get<CompilerCacheEntry>(location);
-                    if (viewStartEntry != null)
+                    var globalFileEntry = _cache.Get<CompilerCacheEntry>(location);
+                    if (globalFileEntry != null)
                     {
-                        // Add the the composite _ViewStart entry as a dependency.
-                        entry.AssociatedViewStartEntry = viewStartEntry;
+                        // Add the the composite _GlobalImport entry as a dependency.
+                        entry.AssociatedGlobalFileEntry = globalFileEntry;
```

Comment:
We have an opportunity to make a perf improvement here - in the previous case compiling _ViewStarts helped since  they'd be used during view execution. Compiling global file entries is less useful since the ordinary case you might not need to execute them at all. We could instead create a combined hash of all applicable _GlobalImports and use that to figure out a difference.